tests/components/arcticspa/test_config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Test the Arctic Spa config flow."""
+from unittest.mock import AsyncMock, patch
+
+from pyarcticspas import SpaResponse
+from pyarcticspas.error import SpaHTTPException, UnauthorizedError
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.arcticspa.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.components.arcticspa import CONF_API_KEY, CONF_API_KEY_VALUE
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "pyarcticspas.Spa.status",
+        return_value=SpaResponse(
+            connected=True,
+        ),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_API_KEY: CONF_API_KEY_VALUE,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["data"] == {
+        CONF_API_KEY: CONF_API_KEY_VALUE,
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pyarcticspas.Spa.status",
+        side_effect=UnauthorizedError(401),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_API_KEY: CONF_API_KEY_VALUE,
+            },
+        )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "pyarcticspas.Spa.status",
+        side_effect=SpaHTTPException(500),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_API_KEY: "fake_api_key",
+            },
+        )
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["errors"] == {"base": "cannot_connect"}
```

Comment:
For these tests, please use parametrize, for a good example of this, checkout the `swiss_public_transport` tests.

Also make sure the tests end either in an abort or a create entry. (So that means patch the lib correctly and test if the config flow can finish)

Reply:
This whole thing was recreated based on the Epion tests. Also, parameterized.